File: framework/ui-elements/src/displays/components/SignText.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractSignText from "../../../gen/displays/components/AbstractSignText";
import SignTextNotifier from "../../../gen/displays/notifiers/SignTextNotifier";
import SignTextRequester from "../../../gen/displays/requesters/SignTextRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import Actionable from "./Actionable";
import { withSnackbar } from 'notistack';

const styles = theme => ({});

class SignText extends AbstractSignText {

	constructor(props) {
		super(props);
		this.notifier = new SignTextNotifier(this);
		this.requester = new SignTextRequester(this);
	};

    sign = (content) => {
        this.requester.signing();
        this.behavior.sign(content, this.state.format, this._successCallback.bind(this), this._failureCallback.bind(this));
    };

}

export default withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SignText));
DisplayFactory.register("SignText", withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SignText)));

File: framework/ui/src/io/intino/alexandria/ui/displays/components/SignText.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.Base64;
import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.notifiers.SignTextNotifier;

import java.nio.charset.StandardCharsets;

public class SignText<DN extends SignTextNotifier, B extends Box> extends AbstractSignText<DN, B> {
    private String signData;

    public SignText(B box) {
        super(box);
    }

    public void text(String content) {
        signData = signMode() == SignMode.CounterSign ? content : signData(content);
    }

    private String signData(String content) {
        return Base64.encode(content.getBytes(StandardCharsets.UTF_8));
    }

    public void execute() {
        sign(signData);
    }
}